Pa11y tests can fail when running in Github Actions

### Issue description
A few of our app modules will fail from time to time during the pa11y tests due to the page crashing. This seems to be a known [issue with puppet running inside of Docker](https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#tips). By default a container will use 64mb of shared memory, (`/dev/shm`) This is not enough memory for chrome at times when puppet runs. Pa11y uses puppet. The `'--disable-dev-shm-usage'` argument can be added to the chromeLaunchConfig in (*I think*) .pa11yci. From what I can see it looks like the options we want would actually be:
```
	"chromeLaunchConfig": {
		"args": [
			"--no-sandbox",
			"--disable-setuid-sandbox",
			"--disable-dev-shm-usage"
		]
	}
```

> **ESTIMATE** TBD

### Steps to reproduce the issue

1. Look at Github actions and see it fail  and the the following error message:
```
Running Pa11y on 15 URLs:
 > http://localhost:3000 - 0 errors
 > http://localhost:3000?cfg=1 - 0 errors
 > http://localhost:3000?cfg=2 - 0 errors
 > http://localhost:3000?cfg=3 - 0 errors
 > http://localhost:3000/expand/A - 0 errors
 > http://localhost:3000/expand/A?cfg=1 - 0 errors
 > http://localhost:3000/expand/K?cfg=2 - 0 errors
 > http://localhost:3000/expand/Y?cfg=3 - 0 errors
(node:2130) UnhandledPromiseRejectionWarning: Error: Page crashed!
    at Page._onTargetCrashed (/home/runner/work/glossary-app/glossary-app/node_modules/puppeteer/lib/Page.js:215:24)
    at CDPSession.<anonymous> (/home/runner/work/glossary-app/glossary-app/node_modules/puppeteer/lib/Page.js:123:56)
    at CDPSession.emit (events.js:314:20)
    at CDPSession._onMessage (/home/runner/work/glossary-app/glossary-app/node_modules/puppeteer/lib/Connection.js:200:12)
    at Connection._onMessage (/home/runner/work/glossary-app/glossary-app/node_modules/puppeteer/lib/Connection.js:112:17)
    at WebSocket.<anonymous> (/home/runner/work/glossary-app/glossary-app/node_modules/puppeteer/lib/WebSocketTransport.js:44:24)
    at WebSocket.onMessage (/home/runner/work/glossary-app/glossary-app/node_modules/puppeteer/node_modules/ws/lib/event-target.js:120:16)
    at WebSocket.emit (events.js:314:20)
    at Receiver.receiverOnMessage (/home/runner/work/glossary-app/glossary-app/node_modules/puppeteer/node_modules/ws/lib/websocket.js:789:20)
    at Receiver.emit (events.js:314:20)
(node:2130) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:2130) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
 > http://localhost:3000/def/a33 - 0 errors
 > http://localhost:3000/def/metastatic - 0 errors
 > http://localhost:3000/def/metastasico?cfg=1 - 0 errors
 > http://localhost:3000/def/acrochordon?cfg=2 - 0 errors
 > http://localhost:3000/def/acrochordon?cfg=3 - 0 errors
 > http://localhost:3000/def/antioncogen?cfg=3 - 0 errors
 > http://localhost:3000/def/a33?cfg=1 - Failed to run

Errors in http://localhost:3000/def/a33?cfg=1:

 • Error: Pa11y timed out (30000ms)

✘ 14/15 URLs passed
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @nciocpl/glossary-app@1.2.2 pa11y: `NODE_ENV=test pa11y-ci`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @nciocpl/glossary-app@1.2.2 pa11y script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2021-03-30T15_41_44_578Z-debug.log
Error: Command failed with exit code 2: npm run pa11y
    at makeError (/home/runner/work/glossary-app/glossary-app/node_modules/start-server-and-test/node_modules/execa/lib/error.js:56:11)
    at handlePromise (/home/runner/work/glossary-app/glossary-app/node_modules/start-server-and-test/node_modules/execa/index.js:114:26)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  command: 'npm run pa11y',
  exitCode: 2,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @nciocpl/glossary-app@1.2.2 pa11y-tests: `start-server-and-test start http://localhost:3000 pa11y`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @nciocpl/glossary-app@1.2.2 pa11y-tests script.
```

### What's the expected result?
- Github actions would not fail.

### What's the actual result?
- It fails from time to time

--- a/.github/workflows/workflow.yml
+++ b/.github/workflows/workflow.yml
@@ -131,91 +131,23 @@ jobs:
           pushd build-artifact
           zip -r ../${BUILD_NAME}.zip *
           popd
-      ## Hack for forcing python3 because using the python shell
-      ## by default uses 2.7
-      - name: Force using python3 for shell
-        shell: bash
-        run: |
-          sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 10
-      ## Install the Akamai netstorage module
-      - name: install python modules
-        shell: bash
-        run: |
-          ## Update Setup Tools so that Edgegrid can install
-          python -m pip install --upgrade pip setuptools wheel
-          pip install -Iv netstorageapi==1.2.12
-          pip install requests edgegrid-python
-      ## Actually do the upload
       - name: Upload artifact to netstorage
-        shell: python
-        env:
-          NS_HOSTNAME: ${{ secrets.ns_hostname }}
-          NS_KEYNAME: ${{ secrets.ns_keyname }}
-          NS_KEY: ${{ secrets.ns_key }}
-          NS_CPCODE: ${{ secrets.ns_cpcode }}
-          PROP_CPCODE: ${{ secrets.prop_cpcode }}
-          EG_HOSTNAME: ${{ secrets.eg_hostname }}
-          EG_CLIENT_TOKEN: ${{ secrets.eg_client_token }}
-          EG_CLIENT_SECRET: ${{ secrets.eg_client_secret }}
-          EG_ACCESS_TOKEN: ${{ secrets.eg_access_token }}
-          ARTIFACT_FILENAME: ${{ format('{0}.zip', steps.set_vars.outputs.build_name) }}
-        run: |
-          import os
-          import requests
-          from akamai.netstorage import Netstorage, NetstorageError
-          from akamai.edgegrid import EdgeGridAuth
-          from urllib.parse import urljoin
-
-          NS_HOSTNAME = os.environ.get('NS_HOSTNAME')
-          NS_KEYNAME = os.environ.get('NS_KEYNAME')
-          NS_KEY = os.environ.get('NS_KEY')
-          NS_CPCODE = os.environ.get('NS_CPCODE')
-          REPO_NAME = os.environ.get('REPO_NAME')
-          BUILD_NAME = os.environ.get('BUILD_NAME')
-          PROP_CPCODE = os.environ.get('PROP_CPCODE')
-          EG_HOSTNAME = os.environ.get('EG_HOSTNAME')
-          EG_CLIENT_TOKEN = os.environ.get('EG_CLIENT_TOKEN')
-          EG_CLIENT_SECRET = os.environ.get('EG_CLIENT_SECRET')
-          EG_ACCESS_TOKEN = os.environ.get('EG_ACCESS_TOKEN')
-
-          artifact_filename = '{0}.zip'.format(BUILD_NAME)
-          local_source = os.path.join(os.getcwd(), artifact_filename)
-          netstorage_destination = '/{0}/{1}/{2}'.format(NS_CPCODE, REPO_NAME, artifact_filename)
-
-          ## Upload to Netstorage
-          ns = Netstorage(NS_HOSTNAME, NS_KEYNAME, NS_KEY, True)
-          ok, response = ns.upload(local_source, netstorage_destination, True)
-
-          if not ok:
-            # Error occurred
-            print(response)
-            exit(1)
-
-          ## Clear cache
-          session = requests.Session()
-          session.auth = EdgeGridAuth(
-            client_token=EG_CLIENT_TOKEN,
-            client_secret=EG_CLIENT_SECRET,
-            access_token=EG_ACCESS_TOKEN
-          )
-
-          ## Purge the cache
-          purge_result = session.post(
-            urljoin(EG_HOSTNAME, '/ccu/v3/delete/cpcode/production'),
-            json={
-              "objects": [NS_CPCODE, PROP_CPCODE]
-            }
-          )
-
-          ## Handle error from purge
-          if (purge_result.status_code != 201) :
-            print("Purge failed")
-            print(purge_result.reason)
-            print(purge_result.content)
-            exit(2)
-
-          ## Sucessfully Completed.
-          print("Uploaded {0} successfully".format(artifact_filename))
-          exit(0)
-
-## TODO: Post notice PR is ready for testing
+        uses: nciocpl/netstorage-upload-action@v1.0.0
+        with:
+          hostname: ${{ secrets.ns_hostname }}
+          cp-code: ${{ secrets.ns_cpcode }}
+          key-name: ${{ secrets.ns_keyname }}
+          key: ${{ secrets.ns_key }}
+          index-zip: true
+          local-path: ${{ format('{0}.zip', env.BUILD_NAME) }}
+          ## Note this action automatically prepends the cpcode to the path.
+          destination-path: ${{ format('/{0}/{1}.zip', env.REPO_NAME, env.BUILD_NAME) }}
+      - name: Clear Site Cache
+        uses: nciocpl/akamai-purge-action@v1.0.2
+        with:
+          hostname: ${{ secrets.eg_hostname }}
+          client-token: ${{ secrets.eg_client_token }}
+          client-secret: ${{ secrets.eg_client_secret }}
+          access-token: ${{ secrets.eg_access_token }}
+          type: 'cpcodes'
+          ref: ${{ format('{0},{1}', secrets.ns_cpcode, secrets.prop_cpcode) }}

--- a/.pa11yci
+++ b/.pa11yci
@@ -4,5 +4,12 @@
     "http://localhost:3000/item/6789",
     "http://localhost:3000/item/99999",
     "http://localhost:3000/chicken"
-  ]
+  ],
+	"chromeLaunchConfig": {
+		"args": [
+			"--no-sandbox",
+			"--disable-setuid-sandbox",
+			"--disable-dev-shm-usage"
+		]
+	}
 }

